shared_kv_storage/db_store/packages/lmdb_store.cpp

```diff
@@ -497,29 +498,37 @@ tcf_err_t db_store::db_store_csv_pop(
     // Resize the valueBuffer to current value size
     valueBuffer.resize(value_size);
 
-    // Fetch the state from the db storage
+    // Fetch the value from the db storage
     result = db_store_get(table, inId.data(), inId.size(), &valueBuffer[0], value_size);
     if (result != TCF_SUCCESS) {
         // SAFE_LOG(TCF_LOG_ERROR, "Failed to get from LMDB database : %d", result);
         return result;
     }
 
     bool isLast = true;
-    // Loop over valueBuffer byte by byte and append to outValue till a ',' is
+    ByteArray first_value;
+    // Loop over valueBuffer byte by byte and append to first_value till a ',' is
     // encountered. Effectively parse the 1st string from the comma separated value.
     for (auto itr : valueBuffer){
         if(itr == ',') {
             isLast = false;
             break;
         }
-        outValue.emplace_back(itr);
+        first_value.emplace_back(itr);
+    }
+
+    // If matching is required and there isn't a match, nothing needs to be done.
+    // Return an error indicating failure in match.
+    if(isMatch && outValue != first_value){
+        return TCF_ERR_SYSTEM;
     }
 
     // If this is the only value, return value and delete key->value from db storage
     if(isLast){
         outValue = valueBuffer;
         return db_store_del(table, inId.data(), inId.size(), valueBuffer.data(), valueBuffer.size());
     }
+    outValue = first_value;
```

Comment:
Please add a comment why valueBuffer is added back. -> ValueBuffer contains remaining csv values  (after deleting the popped value) which is added back